ndcube/tests/test_ndcube_arithmetic.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
+import pytest
+
+import astropy.units as u
+import astropy.wcs
+from astropy.nddata import NDData, StdDevUncertainty
+
+from ndcube import NDCube
+from ndcube.tests.helpers import assert_cubes_equal
+from ndcube.utils.exceptions import NDCubeUserWarning
+
+
+def check_arithmetic_value_and_units(cube_new, data_expected):
+    cube_quantity = u.Quantity(cube_new.data, cube_new.unit)
+    assert u.allclose(cube_quantity, data_expected)
+
+
+@pytest.mark.parametrize('value', [
+    10 * u.ct,
+    u.Quantity([10], u.ct),
+    u.Quantity(np.random.rand(12), u.ct),
+    u.Quantity(np.random.rand(10, 12), u.ct),
+])
+def test_cube_arithmetic_add(ndcube_2d_ln_lt_units, value): # this test methods aims for the special scenario of only integers being added together.
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    # Add
+    new_cube = ndcube_2d_ln_lt_units + value
+    check_arithmetic_value_and_units(new_cube, cube_quantity + value)
+
+
+# Only one of them has a unit.
+# An expected typeError should be raised.
+@pytest.mark.parametrize(("ndc", "value"),
+                        [
+                            ("ndcube_2d_uncertainty_no_unit", NDData(np.ones((10, 12)),
+                                                            wcs=None,
+                                                            unit=u.m,
+                                                            uncertainty=StdDevUncertainty(np.ones((10, 12)) * 0.1))
+                            ),
+                            ("ndcube_2d_ln_lt_units", NDData(np.ones((10, 12)),
+                                                            wcs=None,
+                                                            uncertainty=StdDevUncertainty(np.ones((10, 12)) * 0.1))
+                            ),
+                        ],
+                        indirect=("ndc",))
+def test_arithmetic_add_one_unit(ndc, value):
+    assert isinstance(ndc, NDCube)
+    with pytest.raises(TypeError, match="Adding objects requires that both have a unit or neither has a unit."):
+        ndc + value
+
+
+@pytest.mark.parametrize(("ndc", "value", "expected_kwargs"),
```

Comment:
@PCJY Will you give this function and the `multiply` test below a quick review?  Just take a look and ask questions about anything you don't understand.  If you happen to find any problems, please point those out too.

In summary, I replaced all the `__add__` tests (except `test_arithmetic_add_one_unit`) with this one, which is similar to your `multiply` test below.  That's because the code and logic tree is much simpler than when you first wrote the tests.

I also added an extra case in the parameterizations of your multiply test to catch all the cases.